core/templates/dev/head/filters.js

```diff
@@ -341,26 +341,45 @@ oppia.filter('normalizeWhitespacePunctuationAndCase', [function() {
   };
 }]);
 
-// Note that this filter removes additional new lines or <p><br></p> tags
-// at the end of the string.
-oppia.filter('removeExtraLines', [function() {
+// Filter that targets to remove occurance of blank lines, empty paragraphs
+// and spaces at the end of string. It'll remove <p></p>, <p><br></p>, spaces
+// and &nbsp; occurances from the end of the string.
+oppia.filter('removeExtraLinesAndSpace', [function() {
   return function(string) {
     if (!angular.isString(string)) {
       return string;
     }
     var BLANK_LINES_TEXT = '<p><br></p>';
     var EMPTY_PARA_TEXT = '<p></p>';
     while (1) {
-      var endIndex = string.length;
-      var bStr = string.substring(endIndex - BLANK_LINES_TEXT.length, endIndex);
-      var pStr = string.substring(endIndex - EMPTY_PARA_TEXT.length, endIndex);
-      if (bStr === BLANK_LINES_TEXT) {
-        string = string.substring(0, endIndex - BLANK_LINES_TEXT.length);
-      } else if (pStr === EMPTY_PARA_TEXT) {
-        string = string.substring(0, endIndex - EMPTY_PARA_TEXT.length);
-      } else {
+      var strlength = string.lastIndexOf('<');
```

Comment:
This doesn't seem to really be representing the length of the string, though I'm not really sure what to name this.. perhaps lastIndexOfClosingTag? or similar

Reply:
Cool! I was using this to reduce variable length. :laughing: